### PR TITLE
4.5 - Deprecate newQuery()

### PR DESCRIPTION
**Pull request by markstory**
This completes the backport of the separate query classes to 4.5

### CODE REVIEW

File: src/Database/Connection.php [hunk 1]
```diff
@@ -425,6 +425,13 @@ public function query(string $sql): StatementInterface
      */
     public function newQuery(): Query
     {
+        deprecationWarning(
+            'As of 4.5.0 using query() is deprecated. Instead use `insertQuery()`, ' .
+            '`deleteQuery()`, `selectQuery()` or `updateQuery()`. The query objects ' .
```
Comment (othercorey) [inline src/Database/Connection.php]:
```suggestion
            'As of 4.5.0, using query() is deprecated. Instead, use `insertQuery()`, ' .
            '`deleteQuery()`, `selectQuery()` or `updateQuery()`. The query objects ' .
```

File: src/ORM/Query.php [hunk 2]
```diff
@@ -43,7 +43,7 @@
  * @method \Cake\ORM\Table getRepository() Returns the default table object that will be used by this query,
  *   that is, the table that will appear in the from clause.
  * @method \Cake\Collection\CollectionInterface each(callable $c) Passes each of the query results to the callable
- * @method \Cake\Collection\CollectionInterface sortBy(callable|string $path, int $order = \SORT_DESC, int $sort = \SORT_NUMERIC) Sorts the query with the callback
+ * @method \Cake\Collection\CollectionInterface sortBy(callable|string $path, int $order = \Cake\ORM\::\SORT_DESC, int $sort = \SORT_NUMERIC) Sorts the query with the callback
```
Comment (othercorey) [inline src/ORM/Query.php]:
```suggestion
 * @method \Cake\Collection\CollectionInterface sortBy(callable|string $path, int $order = \SORT_DESC, int $sort = \SORT_NUMERIC) Sorts the query with the callback
```

File: src/Database/Connection.php [hunk 3]
```diff
@@ -425,6 +425,13 @@ public function query(string $sql): StatementInterface
      */
     public function newQuery(): Query
     {
+        deprecationWarning(
+            'As of 4.5.0, using query() is deprecated. Instead, use `insertQuery()`, ' .
```
Comment (ndm2) [inline src/Database/Connection.php]:
```suggestion
            'As of 4.5.0, using newQuery() is deprecated. Instead, use `insertQuery()`, ' .
```

**Comment by othercorey** [comment]
Looks ok except for psalm error.

File: src/Datasource/ConnectionInterface.php [hunk 4]
```diff
@@ -30,15 +30,21 @@
...
  * @method \Cake\Database\StatementInterface query(string $sql) Executes a SQL statement and returns the Statement
  *   object as result. {@see \Cake\Database\Connnection::query()}
+ * @method \Cake\Database\Query\InsertQuery insertQuery(?string $table = null, array $values = [], array $types = [])
+ *   Create a new insert query.
+ * @method \Cake\Database\Query\DeleteQuery deleteQuery(?string $table = null, array $conditions = [], array $types = [])
+ *   Create a new delete query.
+ * @method \Cake\Database\Query\SelectQuery selectQuery($fields = [], $table = [], array $types = [])
+ *   Create a new select query.
+ * @method \Cake\Database\Query\UpdateQuery updateQuery($table = null, array $values = [], array $conditions = [], array $types = [])
+ *   Create a new update query.
```
Comment (othercorey) [inline src/Datasource/ConnectionInterface.php]: This implies a `Database\Connection` is implementing this interface.

Comment (markstory) [reply src/Datasource/ConnectionInterface.php]: oops. I was trying to get psalm to be quiet, but I see that these methods are not part of ConnectionInterface in 5.x so I'll take them out.

File: src/Database/Connection.php [hunk 5]
```diff
@@ -425,6 +425,13 @@ public function query(string $sql): StatementInterface
      */
     public function newQuery(): Query
     {
+        deprecationWarning(
```
Comment (othercorey) [inline src/Database/Connection.php:428]: Is this missing a test?

**Pull request merged**